extensions/kubernetes-client/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesClientProducer.java

```diff
@@ -15,6 +16,7 @@ public class KubernetesClientProducer {
 
     private KubernetesClient client;
 
+    @Named("kubernetes-client")
```

Comment:
Won't it break all the current consumers of `KubernetesClient`?

Reply:
Hmm, actually, that's a good point. I briefly wondered about it and then thought it should be OK but I'm going to check with QOSDK.

Reply:
AFAIU `@Named` serves as a qualifier but not as a _"disqualifier"_, so if no other bean is provided `@Inject` should consider this bean for injection too (but maybe I'm wrong).

I understand that this test verifies this:

https://github.com/quarkusio/quarkus/blob/c9c4946859909bebf42d2fb8bfcdb8433017d230/extensions/kubernetes-client/deployment/src/test/java/io/quarkus/kubernetes/client/deployment/KubernetesClientCDITest.java#L18-L19

Anyway, maybe someone who has better understanding of CDI and ArC can confirm or deny this.

Reply:
OK, if this test passes then it's all good for me.

Reply:
> AFAIU `@Named` serves as a qualifier but not as a _"disqualifier"_, so if no other bean is provided `@Inject` should consider this bean for injection too (but maybe I'm wrong).
> 
> I understand that this test verifies this:
> 
> https://github.com/quarkusio/quarkus/blob/c9c4946859909bebf42d2fb8bfcdb8433017d230/extensions/kubernetes-client/deployment/src/test/java/io/quarkus/kubernetes/client/deployment/KubernetesClientCDITest.java#L18-L19
> 
> Anyway, maybe someone who has better understanding of CDI and ArC can confirm or deny this.

That was my understanding as well but I think that it doesn't stand the reality of tests :)

Reply:
The mentioned test passes (Not an issue with `@Named`).

The IT failed tests are related to:

https://github.com/quarkusio/quarkus/blob/ebee0a4b66d1b4dba9c51165e78d22fbf3b90d3e/integration-tests/openshift-client/src/main/java/io/quarkus/it/openshift/client/Routes.java#L15-L20

Which is yet another thing to consider. Injecting a standard `KubernetesClient` instance is now ambiguous if the OpenShiftClient extension is present.

Unless there's another option, we now need to consider removing the `@DefaultBean` annotation from the `OpenShiftClient` produced by the `OpenShiftClientProducer` and give it a very low priority so that users can still provide their own `OpenShiftClient` bean override. (this would also allow to remove the `@Named` annotation). I originally went for the `@Named` qualifier to be able to preserve the `@DefaultBean`. However, given the current scenario, I don't see any other option other than removing the `@DefaultBean` to fulfill #35383

Reply:
My test only uses `@Inject KubernetesClient client` as well and fails. The "difference" maybe is that we're depending on the `quarkus-openshift-client` extension, not the `quarkus-kubernetes-client` one so maybe that changes things for CDI, since we're getting the client in an indirect way?

Reply:
I was trying to solve this with `@Specializes` annotation, however, I don't think it's working as expected in Quarkus.
Do you have any reference about this?

Reply:
> My test only uses `@Inject KubernetesClient client` as well and fails

The problem has to do with injecting `KubernetesClient` instead of `OpenShiftClient` since then it becomes ambiguous.
It seems that CDI support is limited in Quarkus and not everything seems to work. It's getting hard to solve this, especially when dealing with the `@DefaultBean` problem.

Reply:
Maybe we should just let this as-is? Or request help from CDI experts?

Reply:
I'm trying a few more ideas, if nothing works I'll give up.

Reply:
https://github.com/quarkusio/quarkus/pull/35525#issuecomment-1692765728